src/test/java/org/jenkinsci/remoting/protocol/impl/NetworkLayerTest.java

```diff
@@ -159,193 +99,7 @@ public void doCloseRecv(NetworkLayerFactory serverFactory, NetworkLayerFactory c
         server.get().send(data);
         client.get().awaitByteContent(is(expected));
         assertThat(client.get().asByteArray(), is(expected));
-        server.get().closeRead();
-        client.get().awaitClose();
```

Comment:
I think you should add `client.get().awaitClose()` back to make sure that the client is closed when the server is closed.